lib/detect-testing-library-utils.ts

```diff
@@ -0,0 +1,65 @@
+import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+
+export type DetectionHelpers = {
+  getIsImportingTestingLibrary: () => boolean;
+};
+
+/**
+ * Enhances a given rule `create` with helpers to detect Testing Library utils.
+ */
+export function detectTestingLibraryUtils<
+  TOptions extends readonly unknown[],
+  TMessageIds extends string,
+  TRuleListener extends TSESLint.RuleListener = TSESLint.RuleListener
+>(
+  ruleCreate: (
+    context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
+    optionsWithDefault: Readonly<TOptions>,
+    detectionHelpers: Readonly<DetectionHelpers>
+  ) => TRuleListener
+) {
+  return (
+    context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
+    optionsWithDefault: Readonly<TOptions>
+  ): TRuleListener => {
+    let isImportingTestingLibrary = false;
+
+    // TODO: init here options based on shared ESLint config
```

Comment:
👀 Just in case you missed this.

Reply:
This is fine, it's an actual TODO for next PR where I'll init first shared config.